src/NuGet.Core/NuGet.Packaging/Core/FrameworkNameValidatorUtility.cs

```diff
@@ -34,6 +34,60 @@ internal static bool IsValidFrameworkName(string path)
             return fx != null && fx.Framework != NuGetFramework.UnsupportedFramework.Framework;
         }
 
+        internal static bool IsDottedFrameworkVersion(string path)
+        {
+            foreach (string knownFolder in PackagingConstants.Folders.Known)
```

Comment:
I'm slightly worried about potential noise here. 

Not all folders from `https://github.com/NuGet/NuGet.Client/blob/0e5413751caebf168e7d363c88b9c22a3560fe60/src/NuGet.Core/NuGet.Packaging/PackagingConstants.cs#L19-L30` are relevant. 

For example, `tools` is only relevant for tools packages (.NET Core Tools) and not for any other package type. 
`src`, `analyzers` are also not processed by NuGet.
`content` is packages.config only and the structure doesn't require a tfm. 

I think those 3 at the minimum need skipped